packages/sdk/src/evm/common/any-evm-utils.ts

```diff
@@ -713,59 +860,120 @@ async function encodeConstructorParamsForImplementation(
-        let nativeTokenWrapperAddress =
-          getNativeTokenByChainId(chainId).wrapped.address;
-
-        if (nativeTokenWrapperAddress === ethers.constants.AddressZero) {
-          nativeTokenWrapperAddress = await computeNativeTokenAddress(
-            provider,
-            storage,
-            create2Factory,
-          );
-          infraContracts.push(NativeTokenWrapper.contractType);
-        }
-
-        return nativeTokenWrapperAddress;
-      } else if (p.name && p.name.includes("trustedForwarder")) {
-        if (
-          compilerMetadata.analytics?.contract_name &&
-          compilerMetadata.analytics.contract_name === "Pack"
-        ) {
-          infraContracts.push(EOAForwarder.contractType);
-          const eoaForwarderAddress = await computeEOAForwarderAddress(
-            provider,
-            storage,
-            create2Factory,
-          );
-          return eoaForwarderAddress;
-        }
-        infraContracts.push(Forwarder.contractType);
-        const forwarderAddress = await computeForwarderAddress(
-          provider,
-          storage,
-          create2Factory,
-        );
-        return forwarderAddress;
-      } else {
-        return "";
-      }
-    }),
-  );
+  let constructorParamTypes = constructorParams.map((p) => {
+    if (p.type === "tuple[]") {
+      return ethers.utils.ParamType.from(p);
+    } else {
+      return p.type;
+    }
+  });
+  const constructorParamValues = customParams
+    ? customParams
+    : await Promise.all(
+        constructorParams.map(async (p) => {
+          if (p.name && p.name.includes("nativeTokenWrapper")) {
+            let nativeTokenWrapperAddress =
+              getNativeTokenByChainId(chainId).wrapped.address;
+
+            if (nativeTokenWrapperAddress === ethers.constants.AddressZero) {
+              nativeTokenWrapperAddress = await computeNativeTokenAddress(
+                provider,
+                storage,
+                create2Factory,
+              );
+              infraContracts.push(NativeTokenWrapper.contractType);
+            }
+
+            return nativeTokenWrapperAddress;
+          } else if (p.name && p.name.includes("trustedForwarder")) {
+            if (
+              compilerMetadata.analytics?.contract_name &&
+              compilerMetadata.analytics.contract_name === "Pack"
+            ) {
+              infraContracts.push(EOAForwarder.contractType);
+              const eoaForwarderAddress = await computeEOAForwarderAddress(
+                provider,
+                storage,
+                create2Factory,
+              );
+              return eoaForwarderAddress;
+            }
+            infraContracts.push(Forwarder.contractType);
+            const forwarderAddress = await computeForwarderAddress(
+              provider,
+              storage,
+              create2Factory,
+            );
+            return forwarderAddress;
+          } else {
+            return "";
```

Comment:
this function needs comments + a // TODO here with a warning

i almost wonder if we should throw here, since we don't know what the constructor param value needs to be

Reply:
added a throw here